src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -960,6 +960,15 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 				return Promise.reject(new Error(`Failed to create terminal for task ${task._label}`));
 			}
 
+			this._fireTaskEvent(TaskEvent.start(task, terminal.instanceId, resolver.values));
+			const mapKey = task.getMapKey();
+			this._busyTasks[mapKey] = task;
+			this._fireTaskEvent(TaskEvent.general(TaskEventKind.Active, task, terminal.instanceId));
+
+			const problemMatchers = await this._resolveMatchers(resolver, task.configurationProperties.problemMatchers);
+			const startStopProblemMatcher = new StartStopProblemCollector(problemMatchers, this._markerService, this._modelService, ProblemHandlingStrategy.Clean, this._fileService);
+			this._terminalStatusManager.addTerminal(task, terminal, startStopProblemMatcher);
+
 			let processStartedSignaled = false;
 			terminal.processReady.then(() => {
```

Comment:
The `processReady` promise could be already resolved so the `TaskEvent.processStarted` was fired first